diff --git a/parade.module b/parade.module
index 58745ba..5bf86ec 100644
--- a/parade.module
+++ b/parade.module
@@ -280,27 +280,27 @@ function parade_form_alter(&$form, FormStateInterface $form_state, $form_id) {
   // Modify the Paragraphs field.
   // @todo Check the field formatter instead!
   /** @var \Drupal\Core\Form\FormStateInterface $form */
-  if (isset($form['parade_onepage_sections'])) {
+  if (isset($form['field_paragraphs'])) {
     $form['#attached']['library'][] = 'parade/buttons';
 
-    foreach ($form['parade_onepage_sections']['widget']['add_more'] as $key => $item) {
+    foreach ($form['field_paragraphs']['widget']['add_more'] as $key => $item) {
       if (strpos($key, 'add_more') === FALSE) {
         continue;
       }
       $type = Html::getClass($item['#bundle_machine_name']);
-      $form['parade_onepage_sections']['widget']['add_more'][$key]['#attributes']['class'][] = 'parade-button';
-      $form['parade_onepage_sections']['widget']['add_more'][$key]['#attributes']['class'][] = 'parade-button-' . $type;
+      $form['field_paragraphs']['widget']['add_more'][$key]['#attributes']['class'][] = 'parade-button';
+      $form['field_paragraphs']['widget']['add_more'][$key]['#attributes']['class'][] = 'parade-button-' . $type;
     }
   }
 
   // If it's a form, turn off 'Enable linkedin autofill' checkbox, if
   // 'linkedin_autofill' module isn't installed.
-  if (isset($form['parade_onepage_sections'])) {
+  if (isset($form['field_paragraphs'])) {
     $autofill_module_enabled = \Drupal::moduleHandler()->moduleExists('linkedin_autofill');
 
-    foreach ($form['parade_onepage_sections']['widget'] as $key => $paragraph) {
+    foreach ($form['field_paragraphs']['widget'] as $key => $paragraph) {
       if (is_numeric($key) && isset($paragraph['subform']) && isset($paragraph['subform']['parade_enable_linkedin_autofill']) && $autofill_module_enabled) {
-        $form['parade_onepage_sections']['widget'][$key]['subform']['parade_enable_linkedin_autofill']['#access'] = FALSE;
+        $form['field_paragraphs']['widget'][$key]['subform']['parade_enable_linkedin_autofill']['#access'] = FALSE;
       }
     }
   }
