diff --git a/modules/parade_edit/src/EntityOperations.php b/modules/parade_edit/src/EntityOperations.php
index 7285cf2..519c484 100644
--- a/modules/parade_edit/src/EntityOperations.php
+++ b/modules/parade_edit/src/EntityOperations.php
@@ -56,6 +56,11 @@ class EntityOperations {
       $build['parade_edit_moderation_control'] = $this->formBuilder->getForm(EntityModerationForm::class, $entity);
       $build['parade_edit_moderation_control']['#weight'] = $component['weight'];
     }
+
+    // It's bug? - in hook_preprocess_field the parent object the
+    // isDefaultTranslation is TRUE in all cases, becauses of activeLangcode is
+    // overriden to original lang.
+    $build['parade_onepage_sections']['#object']->parade_edit_isDefaultTranslation = $entity->isDefaultTranslation();
   }
 
 }
